src/store.ts

```diff
@@ -4,7 +4,7 @@ import createStore from 'teaful'
 import type { Profile } from './types/profile'
 
 type Store = {
-	profile: Profile | undefined
+	profile: Partial<Profile> | undefined
```

Comment:
I'm not a huge fan of making `Profile` partial. I'd prefer if the store contained a consistent state. I feel like there's not much point in saving the username and avatar without the wallet, although it might make for a better user experience. In general though, I feel like it would be best to be able to rely on the fact that the store contains a complete profile, so we don't need to check if all those fields are filled in all the time.

Reply:
OK fair point so this means we need to store the avatar & username somewhere before we generate the wallet. I'll give it a try but I would probably do this as a separate PR.